src/System.ServiceModel.Primitives/tests/ServiceModel/MessageContractTest.4.4.0.cs

```diff
@@ -0,0 +1,341 @@
+            mockChannelFactory = new MockChannelFactory<IRequestChannel>(context, new TextMessageEncodingBindingElement().CreateMessageEncoderFactory());
+            mockChannelFactory.OnCreateChannelOverride = (EndpointAddress endpoint, Uri via) =>
+            {
+                // When the channel is created, override the Request method to capture properties and headers
+                mockRequestChannel = (MockRequestChannel)mockChannelFactory.DefaultOnCreateChannel(endpoint, via);
+                mockRequestChannel.RequestOverride = (msg, timeout) =>
+                {
+                    // Echo back a fresh copy of the request
+                    MessageBuffer buffer = msg.CreateBufferedCopy(Int32.MaxValue);
+                    Message replyMsg = buffer.CreateMessage();
+                    return replyMsg;
+                };
+
+                return mockRequestChannel;
+            };
+
+            return mockChannelFactory;
+        };
+
+        MockTransportBindingElement mockBindingElement = new MockTransportBindingElement();
+        mockBindingElement.BuildChannelFactoryOverride = buildFactoryAction;
+
+        CustomBinding binding = new CustomBinding(mockBindingElement);
+        EndpointAddress address = new EndpointAddress("myprotocol://localhost:5000");
+        var factory = new ChannelFactory<IMessageContractRoundTrip_4_4_0>(binding, address);
+        IMessageContractRoundTrip_4_4_0 channel = factory.CreateChannel();
+
+        // Prepare a strongly-typed request
+        request = new RequestBankingData_4_4_0();
+        request.accountName = "My account";
+        request.transactionDate = DateTime.Now;
+        request.requestTestProperty = "test property";
+        request.testValue = "test value";
+        request.testValues = new string[] { "test", "values" };
+        request.testValuesArray = new string[] { "test", "values", "array" };
+
+        // *** EXECUTE *** \\
+        reply = channel.MessageContractRoundTrip(request);
+
+        // *** VALIDATE *** \\
+        Assert.True(String.Equals(request.requestTestProperty, reply.requestTestProperty),
+                    String.Format("TestProperty expected = '{0}' actual = '{1}'",
+                                  request.requestTestProperty, reply.requestTestProperty));
+
+        ValidateArray("TestValues", request.testValues, reply.testValues);
+        ValidateArray("TestValuesArray", request.testValuesArray, reply.testValuesArray);
+    }
+
+    private static void ValidateHeader(Dictionary<string, List<string>> headers, string name, params string[] values)
+    {
+        List<string> headerValues;
+        bool hasHeader = headers.TryGetValue(name, out headerValues);
+        Assert.True(hasHeader, String.Format("Expected header '{0}'", name));
+
+        StringBuilder errorBuilder = new StringBuilder();
+        for (int i = 0; i < headerValues.Count; ++i)
+        {
+            errorBuilder.AppendLine(String.Format("---- Header[{0}] ----", i));
```

Comment:
You can use either AppendFormat or inline string formatting such as `errorBuilder.AppendLine($"---- Header[{i}] ----")`.